scipy/spatial/transform/_rotation.pyx

```diff
@@ -311,23 +311,134 @@ cdef double[:, :] _compute_euler_from_quat(
         # compute first and third angles, according to case
         half_sum = atan2(b, a)
         half_diff = atan2(d, c)
-        
+
         if case == 0:  # no singularities
             _angles[angle_first] = half_sum - half_diff
             _angles[angle_third] = half_sum + half_diff
-        
+
         else:  # any degenerate case
             _angles[2] = 0
             if case == 1:
                 _angles[0] = 2 * half_sum
             else:
                 _angles[0] = 2 * half_diff * (-1 if extrinsic else 1)
-                
+
         # for Tait-Bryan angles
         if not symmetric:
             _angles[angle_third] *= sign
             _angles[1] -= pi / 2
-            
+
+        for idx in range(3):
+            if _angles[idx] < -pi:
+                _angles[idx] += 2 * pi
+            elif _angles[idx] > pi:
+                _angles[idx] -= 2 * pi
+
+        if case != 0:
+            warnings.warn("Gimbal lock detected. Setting third angle to zero "
+                          "since it is not possible to uniquely determine "
+                          "all angles.")
+
+    return angles
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_davenport_from_quat(
+    np.ndarray[double, ndim=2] quat, np.ndarray[double, ndim=1] n1,
+        np.ndarray[double, ndim=1] n2, np.ndarray[double, ndim=1] n3,
+        bint extrinsic=False, bint correct_set=False
```

Comment:
Defaults looks unnecessary?

Reply:
Fair, will remove it. Would maybe be better to force the user to choose between extrinsic and intrinsic too.